huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_plugin.go

```diff
@@ -0,0 +1,320 @@
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("region", region),
+		d.Set("instance_id", instanceID),
+		d.Set("name", name),
+		d.Set("enable", enable),
+		d.Set("version", utils.PathSearch("version", plugin, nil)),
+		d.Set("running", utils.PathSearch("running", plugin, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceDmsRabbitmqPluginDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		deleteRabbitmqPluginHttpUrl = "v2/{project_id}/instances/{instance_id}/rabbitmq/plugins"
+		deleteRabbitmqPluginProduct = "dmsv2"
+	)
+
+	deleteRabbitmqPluginClient, err := cfg.NewServiceClient(deleteRabbitmqPluginProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DMS client: %s", err)
+	}
+	instanceID := d.Get("instance_id").(string)
+	deleteRabbitmqPluginPath := deleteRabbitmqPluginClient.Endpoint + deleteRabbitmqPluginHttpUrl
+	deleteRabbitmqPluginPath = strings.ReplaceAll(deleteRabbitmqPluginPath, "{project_id}", deleteRabbitmqPluginClient.ProjectID)
+	deleteRabbitmqPluginPath = strings.ReplaceAll(deleteRabbitmqPluginPath, "{instance_id}", instanceID)
+	jsonBody := utils.RemoveNil(buildRabbitmqPluginBody(d, false))
+
+	reqOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         jsonBody,
+	}
+
+	retryFunc := func() (interface{}, bool, error) {
+		_, err = deleteRabbitmqPluginClient.Request("PUT", deleteRabbitmqPluginPath, &reqOpt)
+		retry, err := handleMultiOperationsError(err)
+		return nil, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rabbitmqInstanceStateRefreshFunc(deleteRabbitmqPluginClient, instanceID),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 1 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error disabling the plugin: %s", err)
+	}
+
+	// The RabbitMQ disabling plugin is done if the status of its task is SUCCESS.
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"CREATED"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      rabbitmqPluginTaskRefreshFunc(deleteRabbitmqPluginClient, instanceID, d, "rabbitmqPluginModify"),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        1 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the RabbitMQ (%s) disabling plugin (%s) to be done: %s", instanceID, d.Get("name"), err)
+	}
+
+	return resourceDmsRabbitmqPluginRead(ctx, d, cfg)
+}
+
+func rabbitmqPluginTaskRefreshFunc(client *golangsdk.ServiceClient, instanceID string, d *schema.ResourceData,
+	taskName string) resource.StateRefreshFunc {
```

Comment:
as `*schema.ResourceData` is only used to get the plugin name, how about using the name directly?
```suggestion
func rabbitmqPluginTaskRefreshFunc(client *golangsdk.ServiceClient, instanceID, pluginName, taskName string) resource.StateRefreshFunc {
```

Reply:
done